pkg/setting/setting.go

```diff
@@ -206,6 +208,13 @@ type Cfg struct {
 	EnableGzip       bool
 	EnforceDomain    bool
 
+	// GRPC API Server Settings.
```

Comment:
rather than opening up this parallel socket... what about just using the existing one?

https://stackoverflow.com/questions/63668447/why-grpc-go-can-run-grpc-server-and-http-server-at-the-same-address-and-port-bu

Everything seems based on:
```
if strings.Contains(request.Header.Get("Content-Type"), "application/grpc") {
                grpcServer.ServeHTTP(writer, request)
                return
}
```
I do not have a good sense how much overhead that is and if it is a good idea.  I like the idea that it is a simple/consistent place to expose.

Long term (actually not very long... like v8!) using this for more than just grafana live would be very powerful.  It seems as safe/unsafe as giving people access to `/api/*` -- we still need to authenticate and verify connection

Reply:
@bergquist any opinions here?  I like the simplicity that it would just be another flavor API and not require new network configurations.

We can (and should!) have a flag to not start a gRPC server though